x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/fleet/components/agent_reassign_config_flyout/index.tsx

```diff
@@ -0,0 +1,180 @@
+  useCore,
+  useGetAgentConfigs,
+} from '../../../../hooks';
+import { PackageIcon } from '../../../../components/package_icon';
+
+interface Props {
+  onClose: () => void;
+  agentId: string;
+}
+
+export const AgentReassignConfigFlyout: React.FunctionComponent<Props> = ({ onClose, agentId }) => {
+  const { notifications } = useCore();
+  const [selectedAgentConfigId, setSelectedAgentConfigId] = useState<string | undefined>(undefined);
+
+  const agentConfigsRequest = useGetAgentConfigs();
+  const agentConfigs = agentConfigsRequest.data ? agentConfigsRequest.data.items : [];
+
+  const agentConfigRequest = useGetOneAgentConfig(selectedAgentConfigId as string);
+  const agentConfig = agentConfigRequest.data ? agentConfigRequest.data.item : null;
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  async function onSubmit() {
+    try {
+      setIsSubmitting(true);
+      if (!selectedAgentConfigId) {
+        throw new Error('No selected config id');
+      }
+      const res = await sendPutAgentReassign(agentId, {
+        config_id: selectedAgentConfigId,
+      });
+      if (res.error) {
+        throw res.error;
+      }
+      setIsSubmitting(false);
+      const successMessage = i18n.translate(
+        'xpack.ingestManager.agentReassignConfig.successSingleNotificationTitle',
+        {
+          defaultMessage: 'Successfully changed the configuration',
+        }
+      );
+      notifications.toasts.addSuccess(successMessage);
+      onClose();
+    } catch (error) {
+      setIsSubmitting(false);
+      notifications.toasts.addError(error, {
+        title: 'Error',
+      });
+    }
+  }
+
+  return (
+    <EuiFlyout onClose={onClose} size="l" maxWidth={640}>
+      <EuiFlyoutHeader hasBorder aria-labelledby="FleetAgentEnrollmentFlyoutTitle">
+        <EuiTitle size="m">
+          <h2 id="FleetAgentReassigmentFlyoutTitle">
+            <FormattedMessage
+              id="xpack.ingestManager.agentReassignConfig.flyoutTitle"
+              defaultMessage="Assign new agent configuration"
+            />
+          </h2>
+        </EuiTitle>
+        <EuiSpacer size="m" />
+        <EuiText size="s">
+          <FormattedMessage
+            id="xpack.ingestManager.agentReassignConfig.flyoutDescription"
+            defaultMessage="Choose a new agent configuration to assign the selected agent(s) to."
+          />
+        </EuiText>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <EuiFormRow
+              fullWidth
+              label={i18n.translate('xpack.ingestManager.agentReassignConfig.selectConfigLabel', {
+                defaultMessage: 'Agent configuration',
+              })}
+            >
+              <EuiSelect
+                fullWidth
+                options={agentConfigs.map(config => ({
+                  value: config.id,
+                  text: config.name,
+                }))}
+                hasNoInitialSelection
```

Comment:
I would have the currently assigned agent displayed here. Also if i select the agent I am already assigned to, it still attempts to reassign.  They probably should not be allowed to save if they did not select a new configuration. @hbharding ?

Reply:
Going to update 👍

Reply:
it's updated, the default value is the agent already assigned config, and you cannot submit